chain/chunks/src/shards_manager_actor.rs

```diff
@@ -1908,10 +1908,18 @@ impl ShardsManagerActor {
         let forward =
             PartialEncodedChunkForwardMsg::from_header_and_parts(chunk_header, owned_parts);
 
-        let block_producers = self.epoch_manager.get_epoch_block_producers_ordered(&epoch_id)?;
+        let next_epoch_id = self.epoch_manager.get_next_epoch_id(latest_block_hash)?;
+        let this_epoch_block_producers =
+            self.epoch_manager.get_epoch_block_producers_ordered(&epoch_id)?;
+        let next_epoch_block_producers =
+            self.epoch_manager.get_epoch_block_producers_ordered(&next_epoch_id)?;
+        let block_producers: Vec<_> =
+            this_epoch_block_producers.into_iter().chain(next_epoch_block_producers).collect();
```

Comment:
maybe we can add some comment that the loop below already performs deduplication?